pkg/lvmd/cmd/root.go

```diff
@@ -97,31 +99,51 @@ func subMain(ctx context.Context) error {
 	proto.RegisterVGServiceServer(grpcServer, vgService)
 	proto.RegisterLVServiceServer(grpcServer, lvmd.NewLVService(dcm, ocm, notifier))
 	grpc_health_v1.RegisterHealthServer(grpcServer, lvmd.NewHealthService())
-	well.Go(func(ctx context.Context) error {
-		return grpcServer.Serve(lis)
-	})
-	well.Go(func(ctx context.Context) error {
-		<-ctx.Done()
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if err := grpcServer.Serve(lis); err != nil {
+			errCh <- err
+		} else {
+			errCh <- nil
+		}
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		sigCh := make(chan os.Signal, 1)
```

Comment:
`well.Go` does not seem to be handling the signal. Why do you implement this logic? Would you show me where the logic is written about signal.

Reply:
Actually, cybozu-go/well handles signals [here](https://github.com/cybozu-go/well/blob/8cbd22dd58421e97e3cd8a9ecc0ce65000dd682f/signal.go#L28), and `well.Wait` returns `well.errSignaled` when `SIGINT` or `SIGTERM` is raised. The original code of TopoLVM checks if a signal is raised by calling `well.IsSignaled` ([here](https://github.com/topolvm/topolvm/blob/c58ec847f7378509733c0d2b5570149ed625bd59/pkg/lvmd/cmd/root.go#L121)), so it will be necessary to handle signals also in this patch.